components/RecipeDetails.js

```diff
@@ -1,35 +1,54 @@
 import styled from "styled-components";
 import Image from "next/image";
 import Link from "next/link";
+import TrashBin from "public/svg/TrashBin.svg";
+import { useRouter } from "next/router";
+import { v4 as uuidv4, v4 } from "uuid";
+
+export default function RecipeDetails({ recipeDetails, onDeleteRecipe }) {
+  const router = useRouter();
+
+  function handleDeleteAndReroute() {
+    if (confirm("Are you sure you want to delete this recipe?")) {
+      onDeleteRecipe(recipeDetails.id);
+      router.push("/");
+    }
+  }
 
-export default function RecipeDetails({ recipeDetails }) {
   return (
-    <RecipeCard>
-      <h1> {recipeDetails.name}</h1>
-      <StyledImage
-        src={
-          recipeDetails.imageURL
-            ? recipeDetails.imageURL
-            : "/recipe-images/generic-food-3.png"
-        }
-        alt={recipeDetails.name}
-        width={300}
-        height={300}
-      />
-      <p>⏰ {recipeDetails.preparationTime}</p>
-      <ul>
-        {recipeDetails.ingredients.map(
-          (ingredient) =>
-            ingredient.ingredient && (
-              <li key={ingredient.ingredient}>
-                {ingredient.quantity} - {ingredient.ingredient}
-              </li>
-            )
-        )}
-      </ul>
-      <StyledDescription>{recipeDetails.description}</StyledDescription>
-      <StyledLink href="/">⬅ Back</StyledLink>
-    </RecipeCard>
+    <>
+      <RecipeCard>
+        <h1> {recipeDetails.name}</h1>
+        <StyledImage
+          src={
+            recipeDetails.imageURL
+              ? recipeDetails.imageURL
+              : "/recipe-images/generic-food-3.png"
+          }
+          alt={recipeDetails.name}
+          width={300}
+          height={300}
+        />
+        <p>⏰ {recipeDetails.preparationTime}</p>
+        <ul>
+          {recipeDetails.ingredients.map(
+            (ingredient) =>
+              ingredient.name && (
```

Comment:
As you can see in the preview, your code doesn't work anymore! -> Since you changed it to ``ingredient.name`` this will always be ``false`` and the ingredients are no longer rendered.
What Doemser ment: 
Change the name in your data-structure: Each ingredient should have a ``name`` (instead of an ``ingredient``).

Reply:
We changed the data structure inside data.js like suggested. 
We also had to remove the recipes entry inside the local browser storage at the dev tools.